daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/Evaluatable.scala

```diff
@@ -440,6 +441,7 @@ trait ExprEvalMixin[T <: AnyRef]
           }
         }
       }
+    DataValue.assertValueIsNotDataValue(expressionResult)
```

Comment:
This assertions may also not be needed.

Reply:
Given the potential for auto-boxing if you accidently pass a DataValue where a underlying value is wanted, I think the check is worthwhile.